engine/src/main/java/org/pentaho/di/security/ExitInterceptor.java

```diff
@@ -0,0 +1,34 @@
+/*! ******************************************************************************
+ *
+ * Pentaho
+ *
+ * Copyright (C) 2024 by Hitachi Vantara, LLC : http://www.pentaho.com
+ *
+ * Use of this software is governed by the Business Source License included
+ * in the LICENSE.TXT file.
+ *
+ * Change Date: 2029-07-20
+ ******************************************************************************/
+
+package org.pentaho.di.security;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class ExitInterceptor {
+    private static final AtomicBoolean interceptEnabled = new AtomicBoolean(false);
+
+    public static void enableIntercept() {
```

Comment:
Does anything call this method besides our test code?  Do we even need this check?

Reply:
It's "needed" for PanTest.java  due to PanTest calls directly `Pan#main` and Java removed the securityManager class which was used to intercept the System.exit() code in `Pan#main`.

https://github.com/pentaho/pentaho-kettle/blob/c0aaa08dd6950c343e8a048a0ae4b7fbc3177e98/engine/src/test/java/org/pentaho/di/pan/PanTest.java#L215-L220

My thoughts

1. `PanTest.java` isn't a "unit" test, its testing the whole kettle-core + kettle-engine code base it should be moved to our IT test folder for integration folder
2. we should be able to get by with java.lang.ProcessBuilder and check the stdout like we do in the test `PanTest#testListTrans`

Reply:
I am not doubting that the code compiles, just that System.exit is still going happen and stop the running of the all tests. Yesterday, I was looking at quick POC to get around this issue with Java17 and the deprecated SecurityManager class.

Reply:
I At'd the wrong person @SahithiKommagani6 can you provide some clarity on this thread?

Reply:
I missed the inclusion of the  ExitInterceptor code in Pan.java. This will work. I would just add javadoc comment for the class and methods.